lib/AST/TypeCheckRequests.cpp

```diff
@@ -846,6 +846,8 @@ Optional<CaptureInfo> ComputeCaptureInfoRequest::getCachedResult() const {
 }
 
 void ComputeCaptureInfoRequest::cacheResult(CaptureInfo value) const {
+  // FIXME: This is quite messy.  The layout of CaptureInfo could use some
```

Comment:
It's also too large, with a number of fields that in most functions, are null. The capture info should be a single ArrayRef, with the dynamic self and generic captures cases encoded as elements of the array.